classes/UpgradeTools/ModuleAdapter.php

```diff
@@ -182,34 +184,50 @@ public function upgradeModule($id, $name)
     {
         $zip_fullpath = $this->tempPath . DIRECTORY_SEPARATOR . $name . '.zip';
 
-        $addons_url = extension_loaded('openssl')
-            ? 'https://api.addons.prestashop.com'
-            : 'http://api.addons.prestashop.com';
-
-        // Make the request
-        $context = stream_context_create([
-            'http' => [
-                'method' => 'POST',
-                'content' => 'version=' . $this->upgradeVersion . '&method=module&id_module=' . (int) $id,
-                'header' => 'Content-type: application/x-www-form-urlencoded',
-                'timeout' => 10,
-            ],
-        ]);
-
-        // file_get_contents can return false if https is not supported (or warning)
-        $content = Tools14::file_get_contents($addons_url, false, $context);
-        if ($content == false || substr($content, 5) == '<?xml') {
-            return;
-        }
+        $local_module_zip = $this->getLocalModuleZip($name);
+        $local_module_used = false;
+
+        if (null !== $local_module_zip) {
+            try {
+                $filesystem = new Filesystem();
+                $filesystem->copy($local_module_zip, $zip_fullpath);
 
-        if (empty($content)) {
-            $msg = '<strong>' . $this->translator->trans('[ERROR] No response from Addons server.', [], 'Modules.Autoupgrade.Admin') . '</strong>';
-            throw new UpgradeException($msg);
+                $local_module_used = true;
+            } catch (IOException $e) {
+                // Do nothing, we will try to upgrade module from addons
```

Comment:
I'm wondering how will the user know that it didn't work if it will still upgrade using PrestaShop addons...

Maybe the logic is that if he is testing an upgrade then the version he upgrades to doesn't exist anyway so upgrade via addons won't work either ?

Reply:
If something goes wrong with the module upgrade using local source, it will try to get a new version from Addons

Maybe I can add a notice to the autoupgrade logs saying that we will get a new version from Addons ... or I can skip the upgrade of that module

Reply:
I really don't have a definite opinion about how to handle this... if we don't like it later we can still improve, let's not block this nice feature for this